### ISSUE TITLE
ChangeNotification.ResourceData.AdditionalData not being deserialized when using Newtonsoft.Json

### ISSUE THREAD

**Issue by LoungeFlyZ** [microsoftgraph/msgraph-sdk-dotnet#1240]
**Describe the bug**
After deserializing a change notification using Json.Net the ResourceData's AdditionalData collection isn't populated. 

**To Reproduce**
After receiving a change notification use the following code to deserialize it:

`var notifications = JsonConvert.DeserializeObject<ChangeNotificationCollection>(content);`

At this point `notification.ResourceData.AdditionalData` will be null.

**Expected behavior**
That deserialization should work with either System.Text.Json.JsonSerializer, or Newtonsoft.Json.JsonConvert.  Or, it should be documented that parts of the SDK don't work with Newtonsoft.Json.  Given Newtonsoft.Json is the current defacto lib most people use this behavior causes confusion.

**Additional information**
Graph SDK version: 4.17.0 
.Net 6.0

**Comment by andrueastman** [comment]
Hey @LoungeFlyZ,

Thanks for raising this. 

As covered in the upgrade guide [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v4.md#breaking-changes), v4 of the SDK drops NewtonSoft support for System.Text.Json. 

You can always use the inbuilt configured serializer by making a call similar to this.

```cs
var notifications = await _graphServiceClient.HttpProvider.Serializer.DeserializeObject<ChangeNotificationCollection>(content);
```

**Comment by LoungeFlyZ** [comment]
@andrueastman yes I saw the upgrade page.  That made it clear that the SDK uses System.Text.Json internally as well as parameters to various methods JsonDocument.  

However, it doesn't mention at all that deserializing from JSON to something like a ChangeNotificationCollection will not work correctly and will fail silently if you use Newtonsoft.  The code runs perfectly fine, but the resultant object only contained partial data.  Any one used to using Newtonsoft will wonder why their code isn't working correctly, but not expressly failing during deserialization. 

The fix for this is to ensure deserializing using the wrong deserializer throws a not-supported exception. 

In short, if you don't support using another deserializer it should fail.

**Comment by andrueastman** [comment]
Closing this one for now with the moving to the Kiota based versions of the SDK. 

The generated models no longer include annotations directly referencing types/custom converters from System.Text.Json/Json.Net but involves generating explicit serialization/deserialization code into the models (example [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/c4414be6c6c845b6d69f08e2c6ca61548a1f304a/src/Microsoft.Graph/Generated/Models/AadUserConversationMember.cs#L45)).

By providing custom implementations for [IParseNodeFactory](https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/serialization/IParseNodeFactory.cs)/[IParseNode ](https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/serialization/IParseNode.cs)for deserialization and [ISerializationWriterFactory](https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/serialization/ISerializationWriterFactory.cs)/[ISerializationWriter ](https://github.com/microsoft/kiota-abstractions-dotnet/blob/main/src/serialization/ISerializationWriter.cs)for serialization, one may simply register the custom implementations as done in the link below to use with the SDK.

https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/c4414be6c6c845b6d69f08e2c6ca61548a1f304a/src/Microsoft.Graph/Generated/BaseGraphServiceClient.cs#L371
